spyder/widgets/sourcecode/base.py

```diff
@@ -886,61 +886,70 @@ def __move_line_or_selection(self, after_current_line=True):
         cursor = self.textCursor()
         cursor.beginEditBlock()
         start_pos, end_pos = self.__save_selection()
-        add_linesep = False
-        if to_text_string(cursor.selectedText()):

```

Comment:
What happens now if the user has some selected text before trying to move the selection?

It seems like you removed that possibility...